vkubelet/pod.go

```diff
@@ -199,7 +75,17 @@ func (s *Server) createPod(ctx context.Context, pod *corev1.Pod) error {
 	return nil
 }
 
-func (s *Server) deletePod(ctx context.Context, pod *corev1.Pod) error {
+func (s *Server) deletePod(ctx context.Context, namespace, name string) error {
+	// Grab the pod as known by the provider.
+	// Since this function is only called when the Pod resource has already been deleted from Kubernetes, we must get it from the provider so we can call "deletePod".
+	// NOTE: Some providers return a non-nil error in their GetPod implementation when the pod is not found while some other don't.
+	// Hence, we ignore the error and just act upon the pod if it is non-nil (meaning that the provider still knows about the pod).
+	pod, _ := s.provider.GetPod(ctx, namespace, name)
+	if pod == nil {
+		// The provider is not aware of the pod, so we just exit.
```

Comment:
Nit: we should still create the span and make an annotation that the pod didn't exist on it.

Reply:
This could probably be applied in other places as well, and also in general anything that's logged also tends to be a good candidate for adding as an annotation to the span.

Reply:
Taking notes here for working on this during #348.